src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageRestoreBar.xaml.cs

```diff
@@ -158,20 +176,44 @@ public async Task<bool> UIRestorePackagesAsync(CancellationToken token)
             {
                 _packageRestoreManager.PackageRestoreFailedEvent += PackageRestoreFailedEvent;
                 string solutionDirectory = await _solutionManager.GetSolutionDirectoryAsync(token);
-                await _packageRestoreManager.RestoreMissingPackagesInSolutionAsync(solutionDirectory,
-                    this,
-                    new LoggerAdapter(this),
-                    token);
 
-                if (_restoreException == null)
+                if (_isAssetsFileMissing)
                 {
-                    // when the control is first loaded, check for missing packages
-                    await _packageRestoreManager.RaisePackagesMissingEventForSolutionAsync(solutionDirectory, token);
+                    var componentModel = await AsyncServiceProvider.GlobalProvider.GetComponentModelAsync();
+                    _solutionRestoreWorker = new Lazy<ISolutionRestoreWorker>(
```

Comment:
You're not really benefit from this being a lazy. You can just store the worker. 

You can also probably take the worker as an input to the class.